pkg/controller/controller_serviceplan.go

```diff
@@ -56,3 +68,31 @@ func (c *controller) servicePlanDelete(obj interface{}) {
 
 	glog.V(4).Infof("ClusterServicePlan: Received delete event for %v; no further processing will occur", servicePlan.Name)
 }
+
+func (c *controller) reconcileClusterServicePlan(servicePlan *v1beta1.ClusterServicePlan) error {
+	if !servicePlan.Status.RemovedFromBrokerCatalog {
```

Comment:
I would probably rewrite this code into reverse condition:
```go
if servicePlan.Status.RemovedFromBrokerCatalog {
  // handle the deletion
}
```
It allows for handling other scenarios in future.
Otherwise current method can be renamed into `maybeDeleteClusterServicePlan` :)